site/src/sphinx/client-circuit-breaker.rst

```diff
@@ -1,6 +1,270 @@
+
+- ``HALF_OPEN``.
+
+  - After a certain amount of time in the ``OPEN`` state, it changes to the ``HALF_OPEN`` state which tries
+    to find out if the service is available by sending one request.
+    If the request succeeds, the state changes to ``CLOSED``. If it fails, the state changes to ``OPEN``.
+
+Here is the description how it works:
+
+.. uml::
+
+    @startditaa(--no-separation)
+
+                                       +----------------+
+                                       |                |
+                                       |      OPEN      |
+                                       |                |<-------------------+
+                                       +------------+---+      fail again    |
+                                           ^        |                        |
+                                           |        |                        |
+                                           |        |                        |
+                                           |        |                        |
+     under threshold                       |        |                        |
+         +---+                             |        |                        |
+         |   |                             |        |                        |
+         |   v                             |        |                        |
+    +----+-----------+  exceed threshold   |        |     timeout         +--+-------------+
+    |                +---------------------+        +-------------------->|                |
+    |     CLOSED     |                                                    |    HALF_OPEN   |
+    |                |<---------------------------------------------------+                |
+    +----------------+            back to normal(request succeed)         +----------------+
+
+    @endditaa
+
+``CircuitBreakerClient``
+------------------------
+
+Armeria provides two different :api:`Client` implementations depending on the
+:api:`Request` and :api:`Response` types:
+
+- :api:`CircuitBreakerHttpClient`
+- :api:`CircuitBreakerRpcClient`
+
+Let's use :api:`CircuitBreakerHttpClient` to find out what we can do.
+You can use the decorator_ method in :api:`ClientBuilder` to build a :api:`CircuitBreakerHttpClient`:
+
+.. code-block:: java
+
+    import com.linecorp.armeria.client.circuitbreaker.CircuitBreaker;
+    import com.linecorp.armeria.client.circuitbreaker.CircuitBreakerHttpClient;
+    import com.linecorp.armeria.client.circuitbreaker.CircuitBreakerStrategy;
+    import com.linecorp.armeria.client.ClientBuilder;
+    import com.linecorp.armeria.client.HttpClient;
+    import com.linecorp.armeria.common.HttpRequest;
+    import com.linecorp.armeria.common.HttpResponse;
+
+    final CircuitBreakerStrategy<HttpResponse> strategy = CircuitBreakerStrategy.onServerErrorStatus();
+    final CircuitBreaker circuitBreaker = CircuitBreaker.ofDefaultName();
+    final HttpClient client = new ClientBuilder(...)
```

Comment:
We can simplify this by using `HttpClientBuilder`.

Reply:
I used `ClientBuilder` because I wanted to express that the user can build an RPC client if they are using different `Request` and `Response` type even thought I didn't state it explicitly.